crates/experimentation_client/src/lib.rs

```diff
@@ -69,34 +76,98 @@ impl Client {
         }
     }
 
-    pub async fn get_applicable_variant(&self, context: &Value, toss: i8) -> Vec<String> {
-        let experiments: Experiments = self.get_satisfied_experiments(context).await;
+    pub async fn get_applicable_variant(
+        &self,
+        context: &Value,
+        toss: i8,
+    ) -> Result<Vec<String>, String> {
+        let experiments: Experiments = self.get_satisfied_experiments(context).await?;
         let mut variants: Vec<String> = Vec::new();
         for exp in experiments {
             if let Some(v) =
-                self.decide_variant(exp.traffic_percentage, exp.variants, toss)
+                self.decide_variant(exp.traffic_percentage, exp.variants, toss)?
             {
                 variants.push(v.id)
             }
         }
-        variants
+        Ok(variants)
     }
 
-    pub async fn get_satisfied_experiments(&self, context: &Value) -> Experiments {
+    pub async fn get_satisfied_experiments(
+        &self,
+        context: &Value,
+    ) -> Result<Experiments, String> {
         let running_experiments = self.experiments.read().await;
-        running_experiments
+        let filtered_running_experiments = running_experiments
             .iter()
             .filter(|(_, exp)| {
                 jsonlogic::apply(&exp.context, context) == Ok(Value::Bool(true))
             })
             .map(|(_, exp)| exp.clone())
-            .collect::<Experiments>()
+            .collect::<Experiments>();
+
+        if let Some(prefix) = context.get("prefix") {
+            let prefix_list: HashSet<&str> = prefix
+                .as_str()
+                .ok_or_else(|| {
+                    log::error!("Prefix is not a valid string.");
+                    format!("Prefix is not a valid string.")
+                })
+                .map_err_to_string()?
+                .split(",")
+                .collect();
+
+            let prefix_filtered_running_experiments: Vec<Experiment> =
```

Comment:
@mahatoankitkumar this implementation might be inefficient at some point of time, 
can you instead include override_keys in experiment struct and , then we can directly filter out based on that

Reply:
Why inefficient?

Reply:
Why inefficient?

Reply:
its inefficient because you are traversing over all the variants then repeated traversal over overrides of all the variants, 
and unnecessary skip for variants if decodes from_Value to value fails
so adding override_keys in experiment struct would be clean and better

Reply:
Sure will simplify this.

Reply:
But you can't add override_keys in the experiment struct they will be different for different variants.

Reply:
No for all variants inside the same experiment override_keys will be identical , we are enforcing that in api

Reply:
values are different.

Reply:
values can be different , its not relevant in this case since we are filtering on keys of overrides not the values

Reply:
It makes sense to filter on the override_keys already present in Experiments struct

Reply:
We are removing the keys from the overrides if that doesn't match the prefix filter, so will have to traverse all the variants in the overrides to mutate it.